### PR TITLE
Update Candle Stick Visualization 

### PR DESCRIPTION
**Pull request by recca5p**

### CODE REVIEW

File: src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    public void Constructor_InitializesDefaultValues_WhenInstanceIsCreated()
+    {
+        // Act
+        var candleStickVisualization = new CandleStickVisualization();
+
+        // Assert
+        Assert.NotNull(candleStickVisualization);
+        Assert.Equal(ChartType.Candlestick, candleStickVisualization.ChartType);
+        Assert.Null(candleStickVisualization.Title);
+        Assert.Null(candleStickVisualization.DataDefinition);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs:24]: Need to add more assertions for default values

File: src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+        Assert.Equal(ChartType.Candlestick, candleStickVisualization.ChartType);
+        Assert.Equal(dataSourceItem, candleStickVisualization.DataDefinition.DataSourceItem);
+        Assert.Null(candleStickVisualization.Title);
+    }
+
+    [Theory]
+    [InlineData("Stock Prices", null)]
+    [InlineData(null, null)]
+    public void Constructor_SetsTitleAndDataSource_WhenArgumentsAreProvided(string title, DataSourceItem dataSourceItem)
+    {
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs:47]: Missing test cases for Not-null DataSourceItem

Comment (vulh-infragistics) [reply src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs:47]: Also need tests for the case if dataSourceItem.HasTabularData = true / false

File: src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+        var candleStickVisualization = new CandleStickVisualization(title, dataSourceItem);
+
+        // Assert
+        Assert.Equal(title, candleStickVisualization.Title);
+        Assert.Equal(ChartType.Candlestick, candleStickVisualization.ChartType);
+        Assert.Null(candleStickVisualization.DataDefinition);
+    }
+
+    [Fact]
+    public void ChartType_HasCorrectValue_WhenInstanceIsCreated()
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs:58]: Unneeded. It's tested inside the unit tests for Constructors

File: src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+    {
+        // Act
+        var candleStickVisualization = new CandleStickVisualization();
+
+        // Assert
+        Assert.Equal(ChartType.Candlestick, candleStickVisualization.ChartType);
+    }
+
+    [Fact]
+    public void DataDefinition_CanBeUpdated_WhenNewDataSourceItemIsSet()
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs:68]: Remove, it's unneeded

File: src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+            {
+                new DateField("Date"),
+                new NumberField("Spend"),
+                new NumberField("Conversions"),
+                new TextField("Territory")
+            }
+        };
+        excelDataSourceItem.UseExcel("Marketing");
+
+        document.Visualizations.Add(new PivotVisualization("Candle Stick Visualization", excelDataSourceItem)
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs:246]: Check type of the visualization added, Should be CandleStickVisualization

File: src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+                "Values" : [ ],
+                "ShowGrandTotals" : true,
+                "FormatVersion" : 0,
+                "AdHocExpandedElements" : [ ],
+                "Rows" : [ ]
+              }
+            } ]
+            """;
+
+        //Act
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs:214]: Remove

**Review by vulh-infragistics — CHANGES_REQUESTED** [review]
Please check and update the comments for all other classes

File: src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+
+        // Act
+        var candleStickVisualization = new CandleStickVisualization(dataSourceItem);
+
+        // Assert
+        Assert.NotNull(candleStickVisualization);
+        Assert.Equal(ChartType.Candlestick, candleStickVisualization.ChartType);
+        Assert.Equal(dataSourceItem, candleStickVisualization.DataDefinition.DataSourceItem);
+        Assert.Null(candleStickVisualization.Title);
+        Assert.Equal(hasTabularData, candleStickVisualization.DataDefinition.DataSourceItem.HasTabularData);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs:58]: Need assertions to check type of DataDefinition

File: src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+        document.Filters.Add(new DashboardDataFilter("Spend", excelDataSourceItem));
+        document.Filters.Add(new DashboardDateFilter("My Date Filter"));
+
+        RdashSerializer.SerializeObject(document);
+        var json = document.ToJsonString();
+        var actualJson = JObject.Parse(json)["Widgets"];
+        var expected = JArray.Parse(expectedJson);
+
+        var expectedStr = JsonConvert.SerializeObject(expected);
+        var actualStr = JsonConvert.SerializeObject(actualJson);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs:349]: Quite complex if we have expected json -> Parse it -> Convert it to json again. Check it again

**Review by vulh-infragistics — CHANGES_REQUESTED** [review]
Please update the new comments.

File: src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs [hunk 9]
```diff
@@ -0,0 +1,362 @@
...
+                  }
+                } ]
+              }
+            } ]
+            """;
+
+        //Act
+        var document = new RdashDocument("My Dashboard");
+
+        var excelDataSourceItem = new RestDataSourceItem("Marketing Sheet")
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs:305]: Need to re-arrange it. Those codes should not be in // Act part

File: src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs [hunk 10]
```diff
@@ -0,0 +1,362 @@
...
+public class CandleStickVisualizationFixture
+{
+    [Fact]
+    public void Constructor_InitializesDefaultValues_WhenInstanceIsCreated()
+    {
+        // Act
+        var candleStickVisualization = new CandleStickVisualization();
+
+        // Assert
+        Assert.NotNull(candleStickVisualization);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs:21]: Remove this. As the codes below already ensure that candleStickVisualization is not null

File: src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs [hunk 11]
```diff
@@ -0,0 +1,362 @@
...
+        bool hasTabularData)
+    {
+        // Arrange
+        var dataSourceItem = new DataSourceItem { HasTabularData = hasTabularData };
+
+        // Act
+        var candleStickVisualization = new CandleStickVisualization(dataSourceItem);
+
+        // Assert
+        Assert.NotNull(candleStickVisualization);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/CandleStickVisualizationFixture.cs:54]: Remove this. As the codes below already ensure that candleStickVisualization is not null

**Review by vulh-infragistics — CHANGES_REQUESTED** [review]

**Review by hainv-ohio — APPROVED** [review]
